docs/FAQs.md

```diff
@@ -13,6 +13,131 @@ After you set up a repository, you can substitute in any tools you'd like.
+   +dist
+   ```
+
+   - Rather than having to remember to compile each time, we'll update our precommit hook in `.husky/precommit` to build for us on each commit:
+
+   ```diff
+   +pnpm run build
+   +git add dist
+   npx lint-staged
+   ```
+
+   - on the off chance someone isn't running husky, we'll add `.github/workflows/check-dist.yml` ([based on this](https://github.com/actions/typescript-action/blob/main/.github/workflows/check-dist.yml)) to ensure `dist` is up to date:
+
+   ```yml
+   # In TypeScript actions, `dist/` is a special directory. When you reference
+   # an action with the `uses:` property, `dist/index.js` is the code that will be
+   # run. For this project, the `dist/index.js` file is transpiled from other
+   # source files. This workflow ensures the `dist/index.js` file contains the
+   # expected transpiled code.
+   #
+   # If this workflow is run from a feature branch, it will act as an additional CI
+   # check and fail if the checked-in `dist/` directory does not match what is
+   # expected from the build.
+   jobs:
+     check-dist:
+       name: check dist
+       runs-on: ubuntu-latest
+
+       steps:
+         - uses: actions/checkout@v4
+         - uses: ./.github/actions/prepare
+
+         - id: build
+           name: Build dist directory
+           run: pnpm run build
+
+         # This will fail the workflow if the PR wasn't created by Dependabot.
+         - id: diff
+           name: Compare directories
+           run: |
+             if [ "$(git diff --ignore-space-at-eol --text dist/index.js | wc -l)" -gt "0" ]; then
+               echo "Detected uncommitted changes after build."
+               echo "You may need to run 'pnpm run build' locally and commit the changes."
+               echo ""
+               echo "See diff below:"
+               echo ""
+               git diff --ignore-space-at-eol --text dist/index.js
+               echo ""
+               # say this again in case the diff is long
+               echo "You may need to run 'pnpm run build' locally and commit the changes."
+               echo ""
+               exit 1
+             fi
+
+         # If `dist/` was different than expected, and this was not a Dependabot
+         # PR, upload the expected version as a workflow artifact.
+         - id: upload
+           if: ${{ failure() && steps.diff.outcome == 'failure' }}
+           name: Upload artifact
+           uses: actions/upload-artifact@v4
+           with:
+             name: dist
+             path: dist/
+
+   name: Check transpiled index.js in dist is up to date
+
+   on:
+     pull_request: ~
+
+     push:
+       branches:
+         - main
+   permissions:
+     contents: read
+   ```
+
+3. We're going to need an [`action.yml`](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions) file - [here's an example](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action#creating-an-action-metadata-file).
```

Comment:
[Proofreading] The second linked page links to the first linked page itself. I think we can consolidate this item a bit. How about:

```suggestion
3. Create an [`action.yml` metadata file](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action#creating-an-action-metadata-file).
```

Reply:
Sounds good - I want to hit commit suggestion but I think you made it in an old commit

Reply:
Heh silly GitHub - I'll recreate